package/gluon-mesh-batman-adv-core/src/respondd.c

```diff
@@ -212,11 +212,13 @@ static void add_gateway(struct json_object *obj) {
 
 	while (getline(&line, &len, f) >= 0) {
 		char addr[18];
+		char nexthop[18];
 
-		if (sscanf(line, "=> %17[0-9a-fA-F:]", addr) != 1)
+		if (sscanf(line, "=> %17[0-9a-fA-F:] (%*u) %17[0-9a-fA-F:]", addr, &metric, nexthop) != 3) {
```

Comment:
`( %*u)`, according to the sscanf manpage, whitespace characters in the format string match "any amount of white space, including none, in the input."